src/PSRule.Rules.Azure/rules/Azure.AppService.Rule.ps1

```diff
@@ -203,10 +221,59 @@ function global:GetWebSiteConfig {
     param ()
     process {
         $siteConfigs = @(GetSubResources -ResourceType 'Microsoft.Web/sites/config', 'Microsoft.Web/sites/slots/config' | Where-Object {
-            $_.Name -notlike "*/*" -or $_.Name -like "*/web" -or $_.Id -like "*/web"
-        })
+                $_.Name -notlike "*/*" -or $_.Name -like "*/web" -or $_.Id -like "*/web"
+            })
         $siteConfigs;
     }
 }
 
+function global:Get-NodeVersions {
+    <#
+    .SYNOPSIS
+        Get the Node.js versions for the App Service.
+
+    .DESCRIPTION
+        This function retrieves the Node.js versions for the App Service.
+
+    .OUTPUTS
+        Output is a list of Node.js versions used, except the 'NODE|lts' version as this is not version specific,
+        hence not parsable.
+    #>
+    [CmdletBinding()]
+    param ( )
+    
+    [Version[]]$versions = @(
+        # App Service on Linux. Works when main object equals Microsoft.Web/sites or Microsoft.Web/sites/slots
+        $TargetObject.properties.siteConfig.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' }
+        # App Service on Linux. Works for when main object equals Microsoft.Web/sites/config 'web' or Microsoft.Web/sites/slots/config 'web'
+        $TargetObject.properties.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' }
+        # App Service on Linux.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/slots' | 
+        ForEach-Object { $_.properties.siteConfig.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' } }
+        # App Service on Linux.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/config', 'Microsoft.Web/sites/slots/config' |
+        Where-Object name -eq 'web' |
+        ForEach-Object { $_.properties.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' } }
+        
+        # App Service on Windows. Works for when main object equals Microsoft.Web/sites or Microsoft.Web/sites/slots
+        $TargetObject.properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION
+        # App Service on Windows. Works for when main object equals Microsoft.Web/sites/config 'appsettings' or Microsoft.Web/sites/slots/config 'appsettings'
+        $TargetObject.properties.WEBSITE_NODE_DEFAULT_VERSION
+        # App Service on Windows. Works for when main object equals Microsoft.Web/sites/config 'web' or Microsoft.Web/sites/slots/config 'web'
+        $TargetObject.properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION
+        # App Service on Windows.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/slots' |
+        ForEach-Object { $_.properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION }
```

Comment:
When accessed under the `properties.siteConfig.appSettings` this should be an array, with an name/ value object. See documentation example.